rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -3609,5 +3615,74 @@ public Response listStudentSubjectChoices(@PathParam("STUDENTID") Long studentId
     return Response.ok(variableList).build();
   }
 
+  @Path("/students/{STUDENTID:[0-9]*}/studentCard")
+  @GET
+  @RESTPermit(handling = Handling.INLINE)
+  public Response findStudentCard(@PathParam("STUDENTID") Long studentId) {
+    Student student = studentController.findStudentById(studentId);
+    Status studentStatus = checkStudent(student);
+    
+    if (studentStatus != Status.OK)
+      return Response.status(studentStatus).build();
+    
+    User loggedUser = sessionController.getUser();
 
+      if (!loggedUser.getId().equals(student.getId()) && !loggedUser.hasRole(Role.ADMINISTRATOR)) {
+        return Response.status(Status.FORBIDDEN).build();
+      }
+    
+    StudentCard studentCard = studentCardDAO.findByStudent(student);
+    
+    fi.otavanopisto.pyramus.rest.model.StudentCard response = new fi.otavanopisto.pyramus.rest.model.StudentCard();
+    
+    if (studentCard != null) {
+      response.setId(studentCard.getId());
+      response.setActive(studentCard.getActive());
+      response.setExpiryDate(studentCard.getExpiryDate());
+      response.setFirstName(student.getFirstName());
+      response.setLastName(student.getLastName());
+      response.setStudyProgramme(student.getStudyProgramme() != null ? student.getStudyProgramme().getName() : null);
+      response.setType(StudentCardType.valueOf(studentCard.getType().name()));
+      response.setUserEntityId(studentId);
+    }
+    
+    return Response.ok(response).build();
+  }
+  
+  @Path("/students/{STUDENTID:[0-9]*}/studentCard/{CARDID:[0-9]*}")  
+  @PUT
+  @RESTPermit(handling = Handling.INLINE)
+  public Response updateStudentCardActive(@PathParam("STUDENTID") Long studentId, @PathParam("CARDID") Long cardId, Boolean active) {
+
+    // Access
+    User loggedUser = sessionController.getUser();
+    
+    if (!loggedUser.hasRole(Role.STUDENT)) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    Student student = studentController.findStudentById(loggedUser.getId());
```

Comment:
Fetching student and then checking its id seems unnecessary; couldn't you just compare `studentId` to the id of the logged user?